### PR TITLE
Fix #1034 - Give Edge::operator<() weak ordering semantics.

### LINKED ISSUE #1034: std::less on spvtools::opt::Edge is not a weak ordering
This is causing Windows Debug build timeout failures.

### PR DESCRIPTION
**Pull request by dnovillo**
This should fix #1034.  It changes the predicate on operator< to use
label IDs from each block and add them together to define a weak
ordering for std::set.

NOTE: This bug is happening on Windows, I do not have a Windows machine yet, so I'll wait for the bots to finish to make sure that the bug is properly fixed.

### CODE REVIEW

File: source/opt/propagator.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ struct Edge {
   ir::BasicBlock* source;
   ir::BasicBlock* dest;
   bool operator<(const Edge& o) const {
-    return source < o.source || dest < o.dest;
+    return source->id() + dest->id() < o.source->id() + o.dest->id();
```
Comment (dneto0) [inline source/opt/propagator.h]: At first I worried about unsigned overlflow. But that has defined behaviour.  Also, SPIR-V universal limits bound IDs to at most 0x3ffff.

Comment (dnovillo) [reply source/opt/propagator.h]: Yeah, Given the limits on instructions, I figured the id space cannot be bigger than 16 bits.

Comment (atgoo) [reply source/opt/propagator.h]: I don't think edges (2, 4) and (1, 5) should be considered equal. Please use std::pair, std::tie or manual comparisons:
```
if (source < o.source)
  return true;
if (o.source < source)
  return false;
if (dest < o.dest)
  return true;
if (o.dest < dest)
  return false;
```

**Comment by dnovillo** [comment]
Re-based and pushed as 8cfa0c40e0afd5b6fcbedf622fd45afd2de2a055

**Pull request closed without merging**